rusoto/core/src/param.rs

```diff
@@ -9,16 +9,60 @@ pub type Params = BTreeMap<String, Option<String>>;
 
 /// Key:value pair for an service parameter.
 pub trait ServiceParams {
-    fn put(&mut self, key: &str, val: &str);
+    fn put<T: ToParam>(&mut self, key: &str, val: T);
     fn put_key(&mut self, key: &str);
 }
 
 impl ServiceParams for Params {
-    fn put(&mut self, key: &str, val: &str) {
-        self.insert(key.to_owned(), Some(val.to_owned()));
+    fn put<T: ToParam>(&mut self, key: &str, val: T) {
+        self.insert(key.to_owned(), Some(val.to_param()));
     }
 
     fn put_key(&mut self, key: &str) {
         self.insert(key.to_owned(), None);
     }
 }
+
+pub trait ToParam {
```

Comment:
This was added because Lambda has a new `UntagResourceRequest` type that puts a `Vec<String>` into the querystring. Because it was just calling `to_string()` on it before, that failed to compile (`ToString` is not implemented for `Vec<T>`). In addition, this has the benefit of being a small optimization as calling `to_string()` on `&str` is much less efficient than `to_owned()` or `String::from`.